tests/decorators/tests.py

```diff
@@ -466,17 +604,583 @@ def a_view(request):
+        self.assertFalse(hasattr(self.req, 'csrf_processing_done'))
+
+
+class DebugDecoratorsTests(TestCase):
+    """
+    Tests for the debug decorators.
+    """
+    def test_sensitive_variables_decorator(self):
+        @sensitive_variables()
+        def a_func():
+            return 'result'
+        # The decorator takes effect when the function is called
+        result = a_func()
+        self.assertEqual(result, 'result')
+        self.assertEqual(a_func.sensitive_variables, '__ALL__')
+
+    async def test_sensitive_variables_decorator_with_async_function(self):
+        @sensitive_variables()
+        async def an_async_func():
+            return 'result'
+        # The decorator takes effect when the function is called
+        result = await an_async_func()
+        self.assertEqual(result, 'result')
+        self.assertEqual(an_async_func.sensitive_variables, '__ALL__')
+
+    def test_sensitive_post_parameters_decorator(self):
+        @sensitive_post_parameters()
+        def a_view(request):
+            return HttpResponse()
+        req = HttpRequest()
+        resp = a_view(req)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(req.sensitive_post_parameters, '__ALL__')
+
+    async def test_sensitive_post_parameters_decorator_with_async_view(self):
+        @sensitive_post_parameters()
+        async def an_async_view(request):
+            return HttpResponse()
+        req = HttpRequest()
+        resp = await an_async_view(req)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(req.sensitive_post_parameters, '__ALL__')
+
+
+class GzipDecoratorsTests(TestCase):
+    """
+    Tests for the gzip decorator.
+    """
+    # Gzip ignores content that is too short
+    content = "Content " * 100
+
+    def test_gzip_decorator(self):
+        @gzip_page
+        def a_view(request):
+            return HttpResponse(content=self.content)
+        req = HttpRequest()
+        req.META['HTTP_ACCEPT_ENCODING'] = 'gzip'
+        resp = a_view(req)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.headers['Content-Encoding'], 'gzip')
+
+    async def test_gzip_decorator_with_async_view(self):
+        @gzip_page
+        async def an_async_view(request):
+            return HttpResponse(content=self.content)
+        req = HttpRequest()
+        req.META['HTTP_ACCEPT_ENCODING'] = 'gzip'
+        resp = await an_async_view(req)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.headers['Content-Encoding'], 'gzip')
+
+
+class VaryDecoratorsTests(TestCase):
+    """
+    Tests for the vary decorator.
+    """
+    def test_vary_on_headers_decorator(self):
+        @vary_on_headers('Header', 'Another-header')
+        def a_view(request):
+            return HttpResponse()
+        resp = a_view(HttpRequest())
+        self.assertEqual(resp.status_code, 200)
+        self.assertIn('Header, Another-header', resp.headers['Vary'])
```

Comment:
I think it might actually keep the order because it's only every using a set to check new values against existing ones, but it keeps track of (and then recombines) values all within lists. I've changed it anyway since it actually makes for a slightly stronger test.